core-plugin/src/main/groovy/com/github/jrubygradle/internal/gems/GemToIvy.groovy

```diff
@@ -0,0 +1,115 @@
+package com.github.jrubygradle.internal.gems
+
+import com.github.jrubygradle.api.gems.GemInfo
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import groovy.xml.MarkupBuilder
+
+import java.security.MessageDigest
+
+import static com.github.jrubygradle.api.gems.GemVersion.singleGemVersionFromMultipleGemRequirements
+
+/** Converts from Gem metadata to Ivy metadata.
+ *
+ * @author Schalk W. Cronjé
+ *
+ * @since 2.0
+ */
+@CompileStatic
+class GemToIvy {
+
+    /** Create a converter from GEM metadata to Ivy metadata.
+     *
+     * This constructor version assumes that the group is called {@code rubygems}.
+     * @param serverUri URI of the RubyGems server.
+     */
+    GemToIvy(URI serverUri) {
+        this.serverUri = serverUri.toString()
+    }
+
+    /** Create a converter from GEM metadata to Ivy metadata.
+     *
+     * @param serverUri URI of the RubyGems server.
+     * @param group Use a group name other than {@code rubygems}.
+     */
+    GemToIvy(URI serverUri, String group) {
+        this.serverUri = serverUri.toString()
+        this.org = group
+    }
+
+    /** Write the Ivy metadata.
+     *
+     * @param writer Writer for output
+     * @param gem GEM metadata
+     * @return {@code writer} after population with Ivy metadata in XML format.
+     */
+    @CompileDynamic
+    @SuppressWarnings('NoDef')
+    Writer writeTo(Writer writer, GemInfo gem) {
```

Comment:
This is the key code to write an ivy.xml file from Gem information obtained via the APIs.